clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1911,6 +1922,8 @@ private boolean processBackgroundEvents() {
 
                 if (!firstError.compareAndSet(null, e))
                     log.warn("An error occurred when processing the background event: {}", e.getMessage(), e);
+            } finally {
+                kafkaAsyncConsumerMetrics.recordBackgroundEventQueueProcessingTime(time.milliseconds() - startMs);
```

Comment:
Similar to `recordApplicationEventQueueProcessingTime`. The metric description states this is about the time "that the consumer took to process **all available background events**' . Shouldn't we simply take the time from right before the loop to right after it ends, and record the metric once per run of the `processBackgroundEvents`?